src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobExecutionService.java

```diff
@@ -497,4 +527,28 @@ private void deleteCloneRepo(VcsRepositoryUri repositoryUri, @Nullable String co
             throw new LocalCIException(msg, e);
         }
     }
+
+    private void deleteRepoParentFolder(String assignmentRepoCommitHash, Path assignmentRepositoryPath, String testRepoCommitHash, Path testsRepositoryPath) throws IOException {
+        Path assignmentRepo = assignmentRepoCommitHash != null ? Path.of(CHECKED_OUT_REPOS_TEMP_DIR, assignmentRepoCommitHash)
+                : getRepositoryParentFolderPath(assignmentRepositoryPath);
+        FileUtils.deleteDirectory(assignmentRepo.toFile());
+        Path testRepo = testRepoCommitHash != null ? Path.of(CHECKED_OUT_REPOS_TEMP_DIR, testRepoCommitHash) : getRepositoryParentFolderPath(testsRepositoryPath);
+        FileUtils.deleteDirectory(testRepo.toFile());
+    }
+
+    private Path getRepositoryParentFolderPath(Path repoPath) {
+        return repoPath.getParent().getParent();
+    }
+
+    private static String generateCommitHash() {
+        Random random = new Random();
+        StringBuilder hash = new StringBuilder(HASH_LENGTH);
+
+        for (int i = 0; i < HASH_LENGTH; i++) {
+            int index = random.nextInt(CHARACTERS.length());
+            hash.append(CHARACTERS.charAt(index));
+        }
+
+        return hash.toString();
+    }
```

Comment:
That's not really a hashing function, as it doesn't take any input of the commit and just returns a random hex string. Maybe you could check out how Artemis creates the commit hash in other places. Or is there a good reason for just using a random commit independent string?

Reply:
This isn't really meant to be a hash, just a random string, it will only be used as a parent folder for the clone repo, do you think we should have a constant string for this so that build jobs without a hash can be cloned there?

Reply:
A fallback folder sounds better to me 👍

Reply:
Just noticed that we can't actually use a fallback folder since it will be deleted at the end of the buildjob, so if a buildagent is processing multiple build jobs, this will result in a conflict. Changed it instead to UUID.random() since its more readable.